src/app/core/rpc/rpc-state.class.ts

```diff
@@ -38,20 +39,40 @@ export class RPCStateClass {
 
   private blockLoop() {
     if (this.rpc.state.get('blocks') === 0) {
-      setTimeout(this.blockLoop, 1000);
+      setTimeout(this.blockLoop.bind(this), 1000);
     }
     this.rpc.stateCall('getblockchaininfo');
   }
 
   private walletLockedState() {
     this.rpc.state.observe('encryptionstatus')
-      .subscribe(status => this.rpc.state
-        .set('locked', ['Locked', 'Unlocked, staking only'].includes(status)));
+    .subscribe(status => this.rpc.state
+      .set('locked', ['Locked', 'Unlocked, staking only'].includes(status)));
   }
 
-  private initWalletState() {
+  private coldStakeLoop() {
```

Comment:
Will never loop if locked?
This should possibly be called on a unlock event? Maybe subscribe to the locked state

Reply:
It will loop, it just won't do anything if it encrypted.
The rpc command walletsettings sadly requires the wallet to be unlocked..

Currently I have it in a loop, reason being if you modify it from the command line or anything then it will also display.
I could check it once and remember it, but if you have multiple UI's connected to the node then it won't work properly, seems like we could do something with ZMQ there.

Reply:
I'll see if I can subscribe to the unlock event, that's better than looping.